slither/detectors/functions/external_function.py

```diff
@@ -105,9 +107,31 @@ def get_all_function_definitions(base_most_function):
     def function_parameters_written(function):
         return any(p in function.variables_written for p in function.parameters)
 
+    @staticmethod
+    def is_reference_type(parameter):
+        if isinstance(parameter.type, ArrayType):
```

Comment:
Can you add tests that hit each of these conditions i.e. an array (mapping etc.) in storage should not be detected but a memory one should?

Reply:
Sure! Now that you reviewed it, I noticed that is not possible to pass a mapping as a function parameter and passing a struct as a function parameter is only possible after 0.8.0 (or if `pragma experimental ABIEncoderV2` is used).

If you agree, I will remove these lines: https://github.com/crytic/slither/blob/166eaed76178acd7657a773e25b37e68545fb9ff/slither/detectors/functions/external_function.py#L114-L115

and add the test for the structure type.

Reply:
Sounds good to me. I just want to make sure we have coverage for true positives for each case and also true negatives (if the function is already using calldata don't alert)

Reply:
Sorry, I misunderstood. Added true positives in the tests in https://github.com/crytic/slither/pull/1022/commits/2e5aa1a20e88b548698fd2959a215dea6b3f08c0